src/communication/xcWebSocketSubscriber.ts

```diff
@@ -77,40 +76,44 @@ export class DefaultSubscriber implements Subscriber {
         }, heartbeatIntervalSeconds * 1000);
     }
 
-    getModel(xcApiName: string, getModelListener: (compositionModel: CompositionModel) => void): void {
-        let thisSubscriber = this;
-        let command = Commands[Commands.getModel];
-        this.observableMsg
-            .map((rawMessage: MessageEvent) => thisSubscriber.deserializer.deserializeWithoutTopic(rawMessage.data || rawMessage))
-            .filter((data: DeserializedData) => data.command === command)
-            .subscribe((data: DeserializedData) => {
-                log.info("Model " + xcApiName + " received successfully");
-                let compositionModel = thisSubscriber.deserializer.getJsonDataFromGetModelRequest(data.stringData);
-                getModelListener(compositionModel);
-            });
-        let commandData = {
-            Command: command,
-            Data: { "Name": xcApiName }
-        };
-        let input = thisSubscriber.serializer.convertCommandDataToWebsocketInputFormat(commandData);
-        this.webSocket.send(input);
+    getCompositionModel(xcApiName: string): Promise<CompositionModel> {
+        return new Promise((resolve, reject) => {
+            let thisSubscriber = this;
+            let command = Commands[Commands.getModel];
+            this.observableMsg
+                .map((rawMessage: MessageEvent) => thisSubscriber.deserializer.deserializeWithoutTopic(rawMessage.data || rawMessage))
+                .filter((data: DeserializedData) => data.command === command)
+                .subscribe((data: DeserializedData) => {
```

Comment:
use the first() method (as you expect only 1 response)
"first' cleans the observable automatically...

Reply:
done